### PR TITLE
RUN-1211 | Upgrade to jQuery 3.6.1 

### PR DESCRIPTION
**Pull request by mishingo**
**Is this a bugfix, or an enhancement? Please describe.**
Security Enhancement: Address Security issues related to jQuery v1.12.4

**Describe the solution you've implemented**
Update all functions in internal JS files and dependecies for compatibility with jQuery v3.6.1

### CODE REVIEW

File: rundeckapp/grails-app/assets/javascripts/knockout.debug.js [hunk 1]
```diff
@@ -480,7 +480,7 @@ ko.utils = (function () {
                 else
                     throw new Error("The supplied element doesn't support dispatchEvent");
             } else if (useClickWorkaround && element.click) {
-                element.click();
+                element.on('click',);
```
Comment (gschueler) [inline rundeckapp/grails-app/assets/javascripts/knockout.debug.js:483]: seems like it should be `.trigger('click')`?

File: rundeckapp/grails-app/assets/javascripts/ko/handler-urlPathParam.js [hunk 2]
```diff
@@ -35,11 +35,11 @@ ko.bindingHandlers.urlPathParam = {
         var values=[];
         if(typeof(data)!='object'){
             values=[data];
-        }else if(jQuery.isArray(data)){
+        }else if(data.isArray){
```
Comment (gschueler) [inline rundeckapp/grails-app/assets/javascripts/ko/handler-urlPathParam.js:38]: `Array.isArray(data)`?

File: rundeckapp/grails-app/assets/javascripts/ko/handler-urlPathParam.js [hunk 3]
```diff
@@ -35,11 +35,11 @@ ko.bindingHandlers.urlPathParam = {
...
             values=data;
         }else if(typeof(data)=='object'){
             values=ko.utils.unwrapObservable(data['value']);
-            if(!jQuery.isArray(values)){
+            if(!values.isArray){
```
Comment (gschueler) [inline rundeckapp/grails-app/assets/javascripts/ko/handler-urlPathParam.js:42]: also `Array.isArray(values)`?

File: rundeckapp/grails-app/assets/javascripts/resourceModelConfig.js [hunk 4]
```diff
@@ -44,7 +44,7 @@ var ResourceModelConfigControl = Class.create({
     }else{
         content.addClassName('rsrcConfigContent');
     }
-    $(content).select('input').each(function(elem) {
+    $(content).on('select','input').each(function(elem) {
```
Comment (gschueler) [inline rundeckapp/grails-app/assets/javascripts/resourceModelConfig.js:47]: I think the prototype code `$(content).select('input')` should just be converted to `jQuery('input')`

File: rundeckapp/grails-app/assets/javascripts/resourceModelConfig.js [hunk 5]
```diff
@@ -231,7 +231,7 @@ addConfig: function(type) {
 checkForm: function () {
     if ($('configs').down('button.needsSave')) {
         var p=[];
-        $('configs').select('button.needsSave').each(function(e) {
+        $('configs').on('select','button.needsSave').each(function(e) {
```
Comment (gschueler) [inline rundeckapp/grails-app/assets/javascripts/resourceModelConfig.js:234]: I think the prototype code `$(content).select('input')` should just be converted to `jQuery('input')`

File: rundeckapp/grails-app/assets/javascripts/workflow.js [hunk 6]
```diff
@@ -213,7 +213,7 @@ RDWorkflow.parseContextId= function (context) {
         return null;
     }
     //if context is already array, return it
-    if (jQuery.isArray(context)) {
+    if (context.isArray) {
```
Comment (gschueler) [inline rundeckapp/grails-app/assets/javascripts/workflow.js:216]:
```suggestion
    if (Array.isArray(context)) {
```

File: rundeckapp/grails-app/assets/javascripts/workflow.js [hunk 7]
```diff
@@ -230,7 +230,7 @@ RDWorkflow.createContextId = function (contextArr) {
         return null;
     }
     //if context is already array, return it
-    if (!jQuery.isArray(contextArr)) {
+    if (!contextArr.isArray) {
```
Comment (gschueler) [inline rundeckapp/grails-app/assets/javascripts/workflow.js:233]:
```suggestion
    if (!Array.isArray(contextArr)) {
```

File: rundeckapp/grails-app/assets/javascripts/ko/handler-urlPathParam.js (hunk 2 above)
Comment (mishingo) [reply rundeckapp/grails-app/assets/javascripts/ko/handler-urlPathParam.js:38]: yes

File: rundeckapp/grails-app/assets/javascripts/knockout.debug.js (hunk 1 above)
Comment (mishingo) [reply rundeckapp/grails-app/assets/javascripts/knockout.debug.js:483]: it depends click() is native now

**Review by ehe-pd — APPROVED** [review]
Successfully built.
Tested some randomly chosen pages.
The suggestion from Greg's review has been addressed accordingly

**Pull request merged**